plugins/autocomplete/plugin.js

```diff
@@ -644,19 +644,20 @@
 		},
 
 		/**
-		 * Returns the caret position relative to the panel's offset parent.
+		 * Returns the view position relative to the panel's offset parent.
+		 *
+		 * Indicates the start position of the autocomplete dropdown.
 		 * The value returned by this function is passed to the {@link #setPosition} method
 		 * by the {@link #updatePosition} method.
 		 *
+		 * @param {CKEDITOR.dom.range} range The range of text match.
 		 * @returns {Object} Represents the position of the caret.
 		 * @returns {Number} rect.left
 		 * @returns {Number} rect.top
 		 * @returns {Number} rect.bottom
 		 */
-		getCaretRect: function() {
-			var caretClientRect = this.editor.getSelection()
-				.getRanges()[ 0 ]
-				.getClientRects()[ 0 ],
+		getViewStartPosition: function( range ) {
+			var viewPositionRect = range.getClientRects()[ 0 ],
```

Comment:
First rect is good for a generic use case. However there could be a case where the matched word is split into few lines. In that case it's best to start autocomplete from the last range rectangle.

As an example think about autocomplete instance that accepts spaces, and shows autocomplete for "foo bar baz". Consider that you typed "foo bar" where both words are in different lines. Autocomplete should show up for the last rectangle.

![the above "foo bar" case illustrated](https://user-images.githubusercontent.com/5353898/40846160-17f45efa-65b9-11e8-82d3-e681d1a77750.png)